middleware/kubernetes/kubernetes.go

```diff
@@ -207,14 +247,16 @@ func (k *Kubernetes) Records(name string, exact bool) ([]msg.Service, error) {
 
 	nsWildcard := symbolContainsWildcard(namespace)
 	serviceWildcard := symbolContainsWildcard(serviceName)
+	portWildcard := symbolContainsWildcard(port)
+	protocolWildcard := symbolContainsWildcard(protocol)
 
 	// Abort if the namespace does not contain a wildcard, and namespace is not published per CoreFile
 	// Case where namespace contains a wildcard is handled in Get(...) method.
 	if (!nsWildcard) && (len(k.Namespaces) > 0) && (!dnsstrings.StringInSlice(namespace, k.Namespaces)) {
 		return nil, errNsNotExposed
 	}
 
-	k8sItems, err := k.Get(namespace, nsWildcard, serviceName, serviceWildcard, typeName)
+	k8sItems, err := k.Get(namespace, nsWildcard, serviceName, serviceWildcard, endpointname, port, portWildcard, protocol, protocolWildcard, typeName)
```

Comment:
We should probably think of wrapping this stuff up in a struct that holds of of these parameters